splunk-cloudwatch-logs-processor/index.js

```diff
@@ -185,17 +194,27 @@ const CloudWatchToSplunk = (parsed, context, logger, sourcetype, callback) => {
             - Set or remove metadata properties as needed. For descripion of each property, refer to:
             http://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTinput#services.2Fcollector */
 
-            const log = {
-                message: item.message,
-                metadata: {
-                    time: item.timestamp ? new Date(item.timestamp).getTime() / 1000 : Date.now(),
-                    host: parsed.logGroup,
-                    source: parsed.logStream,
-                    sourcetype: sourcetype,
-                    //index: 'main',
-                },
-             };
+            let log;
+
+            if (splunkjson) {
+                try {
+                    log = JSON.parse(item.message);
+                } catch (e) {}
+            }
+
+            if (!log) {
+                log = {
+                    message: item.message,
+                    metadata: {
+                        time: item.timestamp ? new Date(item.timestamp).getTime() / 1000 : Date.now(),
+                        host: parsed.logGroup,
+                        source: parsed.logStream,
+                        sourcetype: sourcetype,
+                        //index: 'main',
+                    },
+                };
 
+            }
```

Comment:
Remove unnecessary blank line:

```suggestion
            }
```

Reply:
The train has left the station. Not pursuing.